db/migrations/20201220213114_create_circle_basic_schema.js

```diff
@@ -0,0 +1,217 @@
+const { baseDown } = require('../utils')
+
+exports.up = async (knex) => {
+  // circle table
+  let table = 'circle'
+  await knex('entity_type').insert({ table })
+  await knex.schema.createTable(table, (t) => {
+    t.bigIncrements('id').primary()
+    t.string('name').notNullable().unique()
+    t.bigInteger('cover').unsigned()
+    t.bigInteger('avatar').unsigned()
+    t.enu('state', ['active', 'banned', 'archived'])
+      .notNullable()
+      .defaultTo('active')
+    t.bigInteger('owner').unsigned().notNullable()
+    t.string('display_name').notNullable()
+    t.text('description')
+    t.enu('provider', ['stripe']).notNullable().defaultTo('stripe')
+    t.string('provider_product_id').notNullable().unique()
+    t.timestamp('created_at').defaultTo(knex.fn.now())
+    t.timestamp('updated_at').defaultTo(knex.fn.now())
+
+    // foreign keys
+    t.foreign('owner')
+      .references('id')
+      .inTable('user')
+      .onUpdate('CASCADE')
+      .onDelete('CASCADE')
```

Comment:
`RESTRICT` ?

Reply:
If restrict, we won't be able to delete an user who has created any circles, for example in implementing user account deletion. However we can replace user data with place holders instead of deleting them. I'll change it to `RESTRICT` for now.